### PR TITLE
Add a test where an action modify one of its dependency

### PR DESCRIPTION
**Pull request by diml**
This PR adds a test exposing the behavior of dune when an action modify one of its dependencies. It seems that the right thing to do would be to forbid that or to sandbox user actions.

### CODE REVIEW

File: test/blackbox-tests/test-cases/action-modifying-a-dependency/dune [hunk 1]
```diff
@@ -0,0 +1,4 @@
+(alias
+ (name   x)
+ (deps   data (universe))
```
Comment (emillon) [inline test/blackbox-tests/test-cases/action-modifying-a-dependency/dune]: This also writes to `data`, so should `data` be a target in this test?

File: test/blackbox-tests/test-cases/action-modifying-a-dependency/run.t [hunk 2]
```diff
@@ -0,0 +1,21 @@
+In this test the "x" alias depends on the file "data" but the action
+associated to "x" appends a line to "data". The current behavior is
```
Comment (emillon) [inline test/blackbox-tests/test-cases/action-modifying-a-dependency/run.t]: It's good to have a neutral description of the behavior, but do you think that's the expected behavior? Is it worth adding a line on what this "should" do instead?

Comment (diml) [reply test/blackbox-tests/test-cases/action-modifying-a-dependency/run.t]: Indeed, I added a description of what it should do.

File: test/blackbox-tests/test-cases/action-modifying-a-dependency/dune (hunk 1 above)
Comment (diml) [reply test/blackbox-tests/test-cases/action-modifying-a-dependency/dune]: That would create a circular dependency

**Comment by emillon** [comment]
That looks great, I think it's OK to merge once the build is fixed (seems unrelated, though).

**Comment by diml** [comment]
The build issue was fixed by #927

**Pull request merged**